### PR TITLE
Add everything

### PR DESCRIPTION
**Pull request by MetRonnie**

### CODE REVIEW

**Comment by oliver-sanders** [comment]
For interest here's another repo which is going the same way about this:

https://github.com/microsoft/TypeScript-TmLanguage

**Comment by oliver-sanders** [comment]
Rolled a TextMate bundle using this grammar and tested in TextMate, look good:

two

Some whitespace is marked as error:

one

**Comment by MetRonnie** [comment]
@oliver-sanders I think it would make sense to tackle this after merging the PR, as then I will be able to include the repo as a submodule in the vscode-cylc repo, so I can resume developing

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "globals": {
+        "Atomics": "readonly",
+        "SharedArrayBuffer": "readonly"
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018
+    },
+    "rules": {
+    }
+}
```
Comment (oliver-sanders) [inline .eslintrc.json:17]: Your text editor really doesn't like newlines at the end of files!

Comment (MetRonnie) [reply .eslintrc.json:17]: Blame eslint for auto-generating it that way! I also noticed that `cylc.tmLanguage.json` doesn't have a newline at EOF... I admit I've only just looked up why that might be a bad thing. Should I push a change with an added newline or is it not necessary?

Comment (oliver-sanders) [reply .eslintrc.json:17]: Might as well fix it, linters will often fail on this.

**Comment by oliver-sanders** [comment]
(sorry forgot to submit the review yesterday)

**Comment by oliver-sanders** [comment]
Evaluating against the native TextMate here's where this Grammar stands:

* [ ] 1.1 (issue with space characters getting marked as error)
* [x] 1.2
* [x] 1.3
* [x] 1.4
* [ ] 2 (works for cylc7 `graph=` rather well, none of the others work for cylc8 anyway)
* Jinja2
  * [x] 3.1
  * [ ] 3.2 (highlighting not universal)
  * [x] 3.3
  * [ ] 3/1.1
  * [x] 3/1.2
  * [x] 3/1.3
* empy
  * [x] 3.1
  * [ ] 3.4
  * [ ] 3.5
  * [ ] 3/1.1
  * [x] 3/1.2
  * [x] 3/1.3

Support is pretty good, especially for a first pass, good job with the ISO8601 work, I wan't expecting anything so advanced, the Pygments lexer doesn't do anything so fancy.

The only things which aught to be fixed before rolling this out are:

* 1.1 the space problem (should be simple).
* 3.2 jinja2 highlighting only performed in sections) see screenshot.

**Pull request merged**